tools/c7n_azure/c7n_azure/session.py

```diff
@@ -69,113 +72,59 @@ def auth_params(self):
         return self._auth_params
 
     def _authenticate(self):
-        try:
-            keyvault_client_id = self._auth_params.get('keyvault_client_id')
-            keyvault_secret_id = self._auth_params.get('keyvault_secret_id')
+        keyvault_client_id = self._auth_params.get('keyvault_client_id')
+        keyvault_secret_id = self._auth_params.get('keyvault_secret_id')
 
-            # If user provided KeyVault secret, we will pull auth params information from it
+        # If user provided KeyVault secret, we will pull auth params information from it
+        try:
             if keyvault_secret_id:
                 self._auth_params.update(
                     json.loads(
                         get_keyvault_secret(keyvault_client_id, keyvault_secret_id)))
+        except HTTPError as e:
+            e.message = 'Failed to retrieve SP credential ' \
+                        'from Key Vault with client id: {0}'.format(keyvault_client_id)
```

Comment:
'User Assigned Identity Client ID: '? (Also would be nice to reflect if MSI used